src/dbnode/integration/wide_query_test.go

```diff
@@ -0,0 +1,334 @@
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"runtime"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/sharding"
+	"github.com/m3db/m3/src/dbnode/storage"
+	"github.com/m3db/m3/src/dbnode/storage/index"
+	"github.com/m3db/m3/src/m3ninx/idx"
+	xclock "github.com/m3db/m3/src/x/clock"
+	"github.com/m3db/m3/src/x/context"
+	xerrors "github.com/m3db/m3/src/x/errors"
+	"github.com/m3db/m3/src/x/ident"
+	xhash "github.com/m3db/m3/src/x/test/hash"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+type shardedIndexChecksum struct {
+	shard     uint32
+	checksums []ident.IndexChecksum
+}
+
+// buildExpectedChecksumsByShard sorts the given IDs into ascending shard order,
+// applying given shard filters.
+func buildExpectedChecksumsByShard(
+	ids []string,
+	allowedShards []uint32,
+	shardSet sharding.ShardSet,
+) []ident.IndexChecksum {
+	shardedChecksums := make([]shardedIndexChecksum, 0, len(ids))
+	for i, id := range ids {
+		checksum := ident.IndexChecksum{ID: []byte(id), Checksum: int64(i)}
+		shard := shardSet.Lookup(ident.StringID(id))
+
+		if len(allowedShards) > 0 {
+			shardInUse := false
+			for _, allowed := range allowedShards {
+				if allowed == shard {
+					shardInUse = true
+					break
+				}
+			}
+
+			if !shardInUse {
+				continue
+			}
+		}
+
+		found := false
+		for idx, sharded := range shardedChecksums {
+			if shard != sharded.shard {
+				continue
+			}
+
+			found = true
+			shardedChecksums[idx].checksums = append(sharded.checksums, checksum)
+			break
+		}
+
+		if found {
+			continue
+		}
+
+		shardedChecksums = append(shardedChecksums, shardedIndexChecksum{
+			shard:     shard,
+			checksums: []ident.IndexChecksum{checksum},
+		})
+	}
+
+	sort.Slice(shardedChecksums, func(i, j int) bool {
+		return shardedChecksums[i].shard < shardedChecksums[j].shard
+	})
+
+	var checksums []ident.IndexChecksum
+	for _, sharded := range shardedChecksums {
+		checksums = append(checksums, sharded.checksums...)
+	}
+
+	return checksums
+}
+
+func TestWideFetch(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow() // Just skip if we're doing a short run
+	}
+
+	var (
+		batchSize     = 7
+		seriesCount   = 100
+		blockSize     = time.Hour * 2
+		verifyTimeout = time.Minute * 2
+	)
+
+	// Test setup
+	idxOpts := namespace.NewIndexOptions().
+		SetEnabled(true).
+		SetBlockSize(time.Hour * 2)
+	nsOpts := namespace.NewOptions().
+		SetIndexOptions(idxOpts).
+		SetRepairEnabled(false).
+		SetRetentionOptions(defaultIntegrationTestRetentionOpts)
+
+	nsID := testNamespaces[0]
+	nsMetadata, err := namespace.NewMetadata(nsID, nsOpts)
+	require.NoError(t, err)
+
+	// Set up file path prefix
+	postfix := atomic.AddUint64(&created, 1) - 1
+	filePathPrefix, err := ioutil.TempDir("", fmt.Sprintf("integration-test-%d", postfix))
```

Comment:
Agreed yeah the `TempDir` already creates a unique tmp dir:
https://golang.org/pkg/io/ioutil/#TempDir
> TempDir creates a new temporary directory in the directory dir. The directory name is generated by taking pattern and applying a random string to the end. If pattern includes a "*", the random string replaces the last "*".

So you can also do "integration-test-*" for the pattern too which would be the same essentially.

Re @arnikola the "this is what is already here for integration tests" we should go rip that out since it's not required.